.github/workflows/build.yml

```diff
@@ -383,15 +453,25 @@ jobs:
         with:
           persist-credentials: false
 
-      - name: Install rust toolchain
-        uses: dtolnay/rust-toolchain@stable
+      # Rust nightly toolchain is needed for test with portable_atomic_util::Arc (see below)
+      - name: Install rust-toolchain - nightly
+        uses: dtolnay/rust-toolchain@nightly
       - name: Install cross (cross-rs) from GitHub
         run: cargo install cross --git https://github.com/cross-rs/cross
       - name: Install bindgen feature & CLI for aws-lc-sys (as needed for many cross targets)
         if: ${{ matrix.target != 'i686-unknown-linux-gnu' }}
         run: cargo add --dev --features bindgen 'aws-lc-sys@>0.20' --package rustls --verbose && cargo install bindgen-cli --verbose
       - run: cross test --package rustls --target ${{ matrix.target }}
 
+      # ############################################################################
+      # TEST with portable_atomic_util::Arc - MUST COME AT THE END OF THIS CI JOB
```

Comment:
I think this comment would be better if it had some mention of why the ordering matters. Presumably because it's changing the source code of the checked out repo?

If we stick to this approach I think I'd mildly prefer to see a stand-alone script that applies the edits, runs the needed tests, and then undoes the edits in an exit trap handler. 

I think there's general aversion to complex bash scripts so this isn't ideal either but it seems better to have it all in one place and somewhat atomic.

Reply:
I have added some info here.

I totally share the adversion to complex shell scripts.

Reply:
I have now removed this whole block. As I said in <https://github.com/rustls/rustls/pull/2285#discussion_r1924001033> I think it should be good enough  to test this scenario in `overwrite-atomic-sync-test`.

We could always add this kind of further testing in the future if needed.